src/lib.rs

```diff
@@ -571,39 +573,39 @@ fn derive_where_internal(attr: TokenStream, item: TokenStream) -> Result<TokenSt
         let trait_ = trait_.path();
 
         // Build necessary generics to construct the implementation item.
-        let (impl_generics, type_generics, mut where_clause) = generics.split_for_impl();
-
-        let default_where = &WhereClause {
-            where_token: Where::default(),
-            predicates: Punctuated::default(),
-        };
-
-        let mut where_clause = where_clause.get_or_insert(default_where).clone();
-
-        // If there are any bounds, insert them into the `where` clause.
-        for generic in &derive_where.generics {
-            where_clause
-                .predicates
-                .push(WherePredicate::Type(match generic {
-                    Generic::CoustomBound(type_bound) => type_bound.clone(),
-                    Generic::NoBound(type_) => {
-                        let mut bounds = Punctuated::default();
-
-                        bounds.push(TypeParamBound::Trait(TraitBound {
-                            paren_token: None,
-                            modifier: syn::TraitBoundModifier::None,
-                            lifetimes: None,
-                            path: trait_.clone(),
-                        }));
-
-                        PredicateType {
+        let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
```

Comment:
Considering we are cloning the `were_clause` in the for-loop now every time, which makes sense, I think this line can be moved outside the for-loop.